packages/garbo/src/resources/mayam.ts

```diff
@@ -0,0 +1,249 @@
+    value: () => 5, // +30% item drop, I don't know how to value that?
+    ring: 1,
+  },
+  {
+    choice: "chair",
+    value: () => 5, // +5 free rests, I don't know how to value that?
+    ring: 1,
+  },
+  {
+    choice: "fur",
+    value: () =>
+      Math.max(
+        0,
+        ...getExperienceFamiliars("free").map(
+          ({ expectedValue }) => expectedValue / 12,
+        ),
+      ) * 100,
+    ring: 1,
+  },
+  {
+    choice: "vessel",
+    value: () => 0, // How do we value MP regen?
+    ring: 1,
+  },
+  {
+    choice: "lightning",
+    value: () => 0,
+    ring: 2,
+  },
+  {
+    choice: "bottle",
+    value: () => 0, // How do we value HP regen?
+    ring: 2,
+  },
+  {
+    choice: "meat",
+    value: () => clamp(myLevel() * 100, 100, 1500),
+    ring: 2,
+  },
+  {
+    choice: "wood",
+    value: () => 0,
+    ring: 2,
+  },
+  {
+    choice: "wall",
+    value: () => 0,
+    ring: 3,
+  },
+  {
+    choice: "cheese",
+    value: () => garboValue($item`goat cheese`),
+    ring: 3,
+  },
+  {
+    choice: "eyepatch",
+    value: () => 0,
+    ring: 3,
+  },
+  {
+    choice: "explosion",
+    value: () => 0,
+    ring: 4,
+  },
+  {
+    choice: "clock",
+    value: () => {
+      if (globalOptions.prefs.valueOfAdventure === undefined) {
+        return 4000 * 5;
+      } else return globalOptions.prefs.valueOfAdventure * 5;
+    },
+    ring: 4,
+  },
+];
+
+const AVAILABLE_RING_OPTIONS: MayamRingSummon[] = MAYAM_RING_OPTIONS.filter(
+  (option) => {
+    return !yamUses().includes(option.choice);
+  },
+);
+
+function findRingValue(choice: string, ring: number): number {
+  const option = AVAILABLE_RING_OPTIONS.find(
+    (option) => option.choice === choice && option.ring === ring,
+  );
+  return option ? option.value() : 0;
+}
+
+function calculateCombinationValue(combination: string[]): number {
+  const combinationString = combination.join(" ");
+  const resonance = findResonance(combinationString);
+  const baseValue = combination
+    .map((choice, index) => findRingValue(choice, index + 1))
+    .reduce((sum, value) => sum + value, 0);
+  return resonance ? baseValue + garboValue(resonance.item) : baseValue;
+}
+
+function generateCombinations(): string[][] {
+  const rings = MayamCalendar.RINGS;
+
+  const combinations: string[][] = [];
+
+  for (const r1 of rings[0]) {
+    for (const r2 of rings[1]) {
+      for (const r3 of rings[2]) {
+        for (const r4 of rings[3]) {
+          combinations.push([r1, r2, r3, r4]);
+        }
+      }
+    }
+  }
+
+  return combinations;
+}
+
+export function findTopCombinations(): string[] {
+  const combinations = generateCombinations();
+  const combinationValues = combinations.map((combination) => ({
+    combination: combination.join(" "),
+    value: calculateCombinationValue(combination),
+  }));
+
+  combinationValues.sort((a, b) => b.value - a.value);
+
+  const selectedCombinations: string[] = [];
+  const usedSymbols: Set<string> = new Set();
```

Comment:
Doesn't this get filled by the first loop? Is it cleared anywhere?

Reply:
Changed all this logic, therefore closing